data/src/new_etl/classes/file_manager.py

```diff
@@ -0,0 +1,262 @@
+import os
+import zipfile
+from datetime import datetime
+from enum import Enum
+from io import BytesIO
+from typing import List
+
+import geopandas as gpd
+from tqdm import tqdm
+
+from config.config import CACHE_FRACTION, ROOT_DIRECTORY
+
+print(f"Root directory is {ROOT_DIRECTORY}")
+
+
+class LoadType(Enum):
+    TEMP = "temp"
+    SOURCE_CACHE = "source_cache"
+    PIPELINE_CACHE = "pipeline_cache"
+
+
+class FileType(Enum):
+    GEOJSON = "geojson"
+    PARQUET = "parquet"
+    PMTILES = "pmtiles"
+    CSV = "csv"
+
+
+class FileManager:
+    """
+    A manager for interacting with cached files or temporary files loaded from or extracting into the local filesystem.
+    """
+
+    _instance = None
+
+    def __init__(self, fraction=CACHE_FRACTION):
+        """
+        Initialize the FileManager with paths for the temporary and cache directories at the root directory of the project.
+        """
+        if FileManager._instance is not None:
+            raise ValueError(
+                "This class is a singleton and has already been initialized"
+            )
+        self.storage_directory = os.path.join(ROOT_DIRECTORY, "storage")
+
+        if not os.path.exists(self.storage_directory):
+            os.makedirs(self.storage_directory)
+
+        self.fraction = fraction
+        self.temp_directory = os.path.join(self.storage_directory, "tmp")
+        self.source_cache_directory = os.path.join(
+            self.storage_directory, "source_cache"
+        )
+        self.pipeline_cache_directory = os.path.join(
+            self.storage_directory, "pipeline_cache"
+        )
+
+        if not os.path.exists(self.temp_directory):
+            os.makedirs(self.temp_directory)
+        if not os.path.exists(self.source_cache_directory):
+            os.makedirs(self.source_cache_directory)
+        if not os.path.exists(self.pipeline_cache_directory):
+            os.makedirs(self.pipeline_cache_directory)
+
+    @staticmethod
+    def get_instance(fraction=CACHE_FRACTION):
```

Comment:
I think this could also be a part of the `__init__` method, right? Like 

```
if FileManager._instance is not None:
    return _instance
```

that way you'd avoid the need for `get_instance` and raising a `ValueError`, and callers don't need to know that `FileManager` is a singleton.